parsers/SG.py

```diff
@@ -267,6 +232,79 @@ def fetch_price(zone_key='SG', session=None, target_datetime=None, logger=None)
     }
 
 
+def __detect_datetime_from_solar_image(solar_image, logger):
+    w, h = solar_image.size
+    crop_left = int(w * 0.75)
+    crop_top = int(h * 0.87)
+    crop_right = int(w * 0.93)
+    crop_bottom = int(h * 0.92)
+    time_img = solar_image.crop((crop_left, crop_top, crop_right, crop_bottom))
+    processed_img = __preprocess_image_for_ocr(time_img)
+    text = image_to_string(processed_img, lang='eng', config='--psm 7 -c tessedit_char_whitelist="0123456789:- "')
+
+    try:
+        time_pattern = r'\d+-\d+-\d+\s+\d+:\d+'
+        time_string = re.search(time_pattern, text, re.MULTILINE).group(0)
+    except AttributeError:
+        msg = 'Unable to get values for SG solar from OCR text: {}'.format(text)
+        logger.warning(msg, extra={'key': 'SG'})
+        return None
+
+    solar_dt = arrow.get(time_string).replace(tzinfo='Asia/Singapore')
+    return solar_dt
+
+
+def __detect_output_from_solar_image(solar_image, logger):
+    w, h = solar_image.size
+    crop_left = int(w * 0.65)
+    crop_top = int(h * 0.74)
+    crop_right = int(w * 0.93)
+    crop_bottom = int(h * 0.80)
+    output_img = solar_image.crop((crop_left, crop_top, crop_right, crop_bottom))
+    processed_img = __preprocess_image_for_ocr(output_img)
```

Comment:
Result after new cropping and pre-processing technique is:
![est-output-preprocess-result](https://user-images.githubusercontent.com/3433964/147286956-8b490e79-cac0-46ca-ade7-c79065027c06.png)